kagami/routes/admin/worker.py

```diff
@@ -0,0 +1,43 @@
+from fastapi import APIRouter, HTTPException
+from ...deps import SupervisorDeps
+
+worker_router = APIRouter(prefix="/admin/worker")
+
+@worker_router.get("/list")
+async def list_workers(supervisor: SupervisorDeps):
+    """get all workers"""
+    workers = supervisor.registered_workers
+    return [
+        {
+            "addr": addr,
+            "providers": [p.name for p in worker.providers.values()],
+            "status": "online" if await supervisor.check_worker_health(addr) else "offline"
```

Comment:
Don't use check_worker_health(). Cost is high and it is a scheduled function for supervisor to use. API should not contain this function.

AND status we have a complete definition in https://github.com/ProjectSoftoid/Kagami/blob/main/kagami/core/models/provider_info.py#L9
worker status is defined by the provider in this worker. You can add worker_status and its maintenance functions in worker_info definition.

Reply:
Another thing is sync your Response model definition on apifox.